ethereum/executionlayer/src/main/java/tech/pegasys/teku/ethereum/executionlayer/ExecutionEngineChannelImpl.java

```diff
@@ -58,37 +56,26 @@ public class ExecutionEngineChannelImpl implements ExecutionEngineChannel {
   private final ExecutionEngineClient executionEngineClient;
   private final Spec spec;
 
-  public static ExecutionEngineChannel create(
-      final String eeEndpoint,
-      final Spec spec,
-      final TimeProvider timeProvider,
-      final Version version,
-      final Optional<String> jwtSecretFile,
-      final Path beaconDataDirectory) {
-    checkNotNull(eeEndpoint);
+  public static ExecutionEngineChannel createImpl(
+      final Optional<Web3JClient> web3JClient, final Version version, final Spec spec) {
     checkNotNull(version);
-    if (eeEndpoint.equalsIgnoreCase(STUB_ENDPOINT_IDENTIFIER)) {
-      EVENT_LOG.executionEngineStubEnabled();
-      return new ExecutionEngineChannelStub(spec, timeProvider, true);
-    }
-    return new ExecutionEngineChannelImpl(
-        createEngineClient(eeEndpoint, timeProvider, version, jwtSecretFile, beaconDataDirectory),
-        spec);
+    checkArgument(web3JClient.isPresent());
```

Comment:
If web3jclient is required we should just make the param not be an optional.

Reply:
it would require to have dependency at the caller level. Actually a lot of decisions in this pr were affected by module dependencies